tests/foreman/api/test_host.py

```diff
@@ -1270,6 +1270,211 @@ def test_positive_list_hosts_thin_all(module_target_sat):
     assert 'architecture' not in keys
 
 
+def _create_transient_packages(target_sat, host, package_data):
+    """
+    Helper function to create transient packages for a host in the database.
+
+    Args:
+        target_sat: Satellite instance
+        host: Host object to associate packages with
+        package_data: List of package dictionaries with name, version, release, arch
+
+    Returns:
+        None
+
+    Raises:
+        AssertionError: If package ID cannot be parsed from database response
+    """
+    for pkg in package_data:
+        epoch = pkg.get('epoch', '0')  # Default epoch to 0 if not specified
+        nvra = f"{pkg['name']}-{pkg['version']}-{pkg['release']}.{pkg['arch']}"
+        nvrea = f"{epoch}:{pkg['name']}-{pkg['version']}-{pkg['release']}.{pkg['arch']}"
+
+        # Insert package record using psql with tabular output format
+        sql_cmd = (
+            f"INSERT INTO katello_installed_packages "
+            f"(name, nvra, nvrea, version, release, arch, epoch) "
+            f"VALUES ('{pkg['name']}', '{nvra}', '{nvrea}', '{pkg['version']}', "
+            f"'{pkg['release']}', '{pkg['arch']}', '{epoch}') "
+            f"ON CONFLICT DO NOTHING;"
+        )
+        target_sat.execute(f'sudo -u postgres psql foreman -c "{sql_cmd}"')
```

Comment:
Could we leverage the `target_sat.query_db` helper here and bellow? (with or without json `output_format` if needed) 
```suggestion
        target_sat.query_db(sql_cmd)
```

Reply:
ah I missed this, yes I'll use the helper. I had a feeling we had one...